sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1258,6 +1258,14 @@ object SQLConf {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefault(10L)
 
+  val STREAMING_STOP_TIMEOUT =
+    buildConf("spark.sql.streaming.stopTimeout")
+      .doc("How long to wait for the streaming execution thread to stop when calling the " +
+        "streaming query's stop() method in milliseconds. 0 means to wait infinitely. Setting a " +
```

Comment:
I'd assert the value in configuration side instead of throwing IllegalArgumentException when calling stop() - that fits better for "fail-fast".

Reply:
Sometimes it would fail in a case where it is very hard to debug though. A creation of a SparkSession may fail because of the poorly set parameters, which makes it harder to debug. Therefore we're a lot more lenient with the setting of bad configurations.

Reply:
I'm sorry, but I'm not sure I got your point about "harder to debug". `checkValue` can leave an error message, and it also throws IllegalArgumentException. The error message would be clearer, as we don't modify the error message of IllegalArgumentException given from Thread.join() which doesn't explain about the configuration.

Also while stop() will throw IllegalArgumentException, while the method has no parameter which is feeling odd (given the exception comes from Thread.join with parameter), and the method still does the thing even it throws IllegalArgumentException - that does interrupt the thread. It's effectively not different with TimeoutException. If we still want to throw exception when stop is called, I'd rather check the value and don't call join and throw TimeoutException directly - it will at least explain about bad configuration.